### ISSUE TITLE
ERROR TypeError: Cannot read property 'host' of undefined

### ISSUE THREAD

**Issue by Hugofromfrance** [dappsnation/akita-ng-fire#43]
Hi,

I recently tried to use akita-ng-fire
I met this issue.

```
core.js:7187 ERROR TypeError: Cannot read property 'host' of undefined
    at Object.push../node_modules/@firebase/firestore/dist/index.cjs.js.Firestore.configureClient (index.cjs.js:21069)
    at Object.push../node_modules/@firebase/firestore/dist/index.cjs.js.Firestore.ensureClientConfigured (index.cjs.js:21061)
    at Object.get [as _isTerminated] (index.cjs.js:21015)
    at datorama-akita.js:481
    at Array.forEach (<anonymous>)
    at deepFreeze (datorama-akita.js:474)
    at datorama-akita.js:484
    at Array.forEach (<anonymous>)
    at deepFreeze (datorama-akita.js:474)
    at datorama-akita.js:484
```

I followed this: https://dev.to/arielgueta/getting-started-with-akita-and-firebase-3pe2
I really don't understand why i face this issue.

If you have an idea i 'll appreciate.

I tried to user akita without akita-ng-fire, it works. As soon as i request firestore and try to set the store with response, i got this error.

Thanks.

**Comment by Hugofromfrance** [comment]
Hi @GrandSchtroumpf I found the problem.

The problem occurs when I try to use syncCollection with documents that contain "reference" type fields. When I delete references from documents, everything works perfectly.

Do you know if this kind on field is ok with akita-ng-fire or with Akita ?

Thanks

**Comment by GrandSchtroumpf** [comment]
Thanks for reporting. I've never used a reference object. How does it look like when you get it from `valueChanges()` ? Is is an object or a simple string ?

**Comment by Hugofromfrance** [comment]
It is an object of type `DocumentReference`

**Comment by GrandSchtroumpf** [comment]
My guess it's because `DocumentReference` is a class and Akita supports only flat object. You can read more about this : 
https://netbasal.gitbook.io/akita/general/class-support

**Comment by Hugofromfrance** [comment]
Thanks for the advice, il will try this way

**Comment by dylanwatsonsoftware** [comment]
I think I accidentally reproduced your issue in https://github.com/datorama/akita/issues/354. I'll look at how I might be able extend akita-ng-fire to achieve my own nefarious aspirations this week if I can.

I think `akitaPreAddEntity` doesn't entirely solve my problems, as ideally I'd be able to create documents with DocumentReferences too. I'll have to have a think.

**Comment by dylanwatsonsoftware** [comment]
I've posted how I solved my issue here: https://github.com/datorama/akita/issues/354#issuecomment-569207708

Hopefully it helps!

**Comment by Maxstgt** [comment]
> I've posted how I solved my issue here: [datorama/akita#354 (comment)](https://github.com/datorama/akita/issues/354#issuecomment-569207708)
> I've posted how I solved my issue here: datorama/akita#354 (comment)
> Hopefully it helps!

Awesome, thanks!
